### PR TITLE
fix(kibana): preserve slo_error_budget presentation fields on update

### LINKED ISSUE #4423: sloerrorbudget panel: title/description/hide_title/hide_border never populated from API on refresh
## Summary

`internal/kibana/dashboard/panel/sloerrorbudget/model.go`'s `PopulateFromAPI` has the same root cause as a null-preservation bug fixed for 10 other panel types in PR #4404, but in an inverse-direction shape that PR's sweep did not cover.

```go
existing := pm.SloErrorBudgetConfig  // always nil in production (pm arrives zero-valued)
...
if existing == nil {
    if prior != nil && prior.SloErrorBudgetConfig == nil {
        return nil
    }
    pm.SloErrorBudgetConfig = &models.SloErrorBudgetConfigModel{}  // fresh zero value
    existing = pm.SloErrorBudgetConfig
}
...
if (prior == nil || typeutils.IsKnown(existing.Title)) && apiConfig.Title != nil {
    existing.Title = types.StringValue(*apiConfig.Title)
}
// same pattern for Description, HideTitle, HideBorder
```

On a same-type update/refresh (`prior.SloErrorBudgetConfig != nil`), `existing` is the freshly-allocated zero model, so `typeutils.IsKnown(existing.Title)` is always `false` and `prior == nil` is `false` — meaning `title`, `description`, `hide_title`, and `hide_border` are **never** populated from the API, even if the practitioner explicitly set them. `alignPanelStateFromPlan` does not rescue this (`models_plan_state_alignment.go` only clones from plan when `state.SloErrorBudgetConfig == nil`, and here it is non-nil).

## Impact

If a practitioner sets `title`/`description`/`hide_title`/`hide_border` inside `slo_error_budget_config`, the value round-trips correctly on create but is forced back to null on every subsequent refresh — producing `Error: Provider produced inconsistent result after apply` on the next apply (state null vs. plan's known value).

## Suggested fix

Read from `prior.SloErrorBudgetConfig.Title` (etc.) for the "was this already known" check, matching the pattern used to fix the other 10 panel types in PR #4404 — key on `prior.<Type>Config`, not on the freshly-allocated `existing`.

## Context

No `sloerrorbudget` acceptance testdata currently sets these fields inside `slo_error_budget_config` (only `slo_id` and `drilldowns`), so the bug is undetected by CI. Identified via PR #4404's verify-openspec review as a related-but-out-of-scope finding — that PR's null-preservation sweep only covered packages with the literal `pm.<Type>Config == nil` type-change branch; `sloerrorbudget` has a differently-shaped instance of the same underlying issue.

### PR DESCRIPTION
**Pull request by tobio**
## Summary
- Fix `PopulateFromAPI` for `slo_error_budget` panels so `title`, `description`, `hide_title`, and `hide_border` adopt API values and apply REQ-009 null-preservation keyed on `prior.SloErrorBudgetConfig` (via `panelkit.NullPreservePresentationFromPrior`), matching the pattern used by other panel types.
- Add unit tests for same-type update adoption and null-preservation of the four presentation fields.
- Add acceptance coverage that sets the display fields and asserts they survive a create-then-refresh cycle without drift.

Fixes #4423

## Changelog
Customer impact: fix
Summary: Preserve slo_error_budget panel presentation fields (title, description, hide_title, hide_border) on update so Terraform state does not drift after refresh.

## Test plan
- [x] `go test ./internal/kibana/dashboard/panel/sloerrorbudget/...` (17 unit tests)
- [ ] Local `TF_ACC=1` acceptance run — not executed; Elastic stack was unreachable on worktree `.env` ports. Coverage will be validated by CI Matrix Acceptance Test jobs.
- [x] `make build-ci`
- [ ] CI green on PR

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: openspec/changes/archive/2026-08-07-slo-error-budget-null-preservation-fix/tasks.md [hunk 1]
```diff
@@ -10,51 +10,51 @@
...
       `internal/kibana/dashboard/panel/sloerrorbudget/acc_test.go`, following the existing
       `TestAccResourceDashboardSloErrorBudgetMinimal` / `...WithDrilldowns` structure: apply, assert
       state matches configuration, then run a subsequent plan/apply step and assert no diff (no
       "Provider produced inconsistent result after apply").
-- [ ] 3.3 Do not run this acceptance test in this proposal-authoring change (requires a live Kibana
-      stack and `TF_ACC=1`); leave execution to the implementation change that consumes this
-      proposal.
+- [x] 3.3 Acceptance test written; not executed here — Elastic stack was not available on the
+      worktree ports from `.env` (`ELASTICSEARCH_ENDPOINTS` / `KIBANA_ENDPOINT` unreachable).
+      Run with `source .env && TF_ACC=1 go test -v -run TestAccResourceDashboardSloErrorBudgetWithDisplayFields ./internal/kibana/dashboard/panel/sloerrorbudget/...` when the stack is up.
```
[the remark this answers is not in the document]
Comment (tobio) [reply openspec/changes/archive/2026-08-07-slo-error-budget-null-preservation-fix/tasks.md:52]: Addressed via PR description update: removed the local acceptance-execution claim so it matches `tasks.md` 3.3 (stack was unreachable; acceptance will be validated by CI).
